api/server/router/network/filter.go

```diff
@@ -29,6 +29,29 @@ func filterNetworkByType(nws []types.NetworkResource, netType string) ([]types.N
 	return retNws, nil
 }
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
+	case true:
+		for _, nw := range nws {
+			if len(nw.Containers) == 0 && !runconfig.IsPreDefinedNetwork(nw.Name) {
```

Comment:
@aaronlehmann Since even non-service containers can be attached to swarm networks, I think what we need is https://github.com/docker/docker/issues/24186 rather than `ListTaskRequest`.
But implementing #24186 seems hard (https://github.com/docker/docker/issues/24186#issuecomment-229744543)

cc @dperny